db/view/view.cc

```diff
@@ -2973,6 +2974,9 @@ class view_builder::consumer {
 
     stop_iteration consume_end_of_partition() {
         inject_failure("view_builder_consume_end_of_partition");
+        if (utils::get_local_injector().enter("view_builder_consume_end_of_partition_delay_3s")) {
+            seastar::sleep(std::chrono::seconds(3)).get();
+        }
```

Comment:
This isn't specifically a comment about this patch, but: I'm not happy about the trend of having more and more of these injections. Here we have a two-line function, with two different injections :-(